services/src/pro/datasets/external/copernicus_dataspace/ids.rs

```diff
@@ -0,0 +1,429 @@
+}
+
+impl Sentinel2Product {
+    // TODO: move to sentinel2 to separate concerns
+    pub fn product_type(&self) -> &str {
+        match self {
+            Self::L1C => "S2MSI1C",
+            Self::L2A => "S2MSI2A",
+        }
+    }
+
+    pub fn main_file_name(&self) -> &str {
+        match self {
+            Self::L1C => "MTD_MSIL1C.xml",
+            Self::L2A => "MTD_MSIL2A.xml",
+        }
+    }
+
+    pub fn driver_name(&self) -> &str {
+        match self {
+            Self::L1C => "SENTINEL2_L1C",
+            Self::L2A => "SENTINEL2_L2A",
+        }
+    }
+}
+
+// exemplary gdalinfo output:
+// $ gdalinfo /vsizip/download/S2A.zip/S2A_MSIL2A_20160103T154242_N0201_R068_T17NRA_20160103T154241.SAFE/MTD_MSIL2A.xml
+// [...]
+//     Subdatasets:
+//     SUBDATASET_1_NAME=SENTINEL2_L2A:/vsizip/download/S2A.zip/S2A_MSIL2A_20160103T154242_N0201_R068_T17NRA_20160103T154241.SAFE/MTD_MSIL2A.xml:10m:EPSG_32617
+//     SUBDATASET_1_DESC=Bands B2, B3, B4, B8 with 10m resolution, UTM 17N
+//     SUBDATASET_2_NAME=SENTINEL2_L2A:/vsizip/download/S2A.zip/S2A_MSIL2A_20160103T154242_N0201_R068_T17NRA_20160103T154241.SAFE/MTD_MSIL2A.xml:20m:EPSG_32617
+//     SUBDATASET_2_DESC=Bands B5, B6, B7, B8A, B11, B12, AOT, CLD, SCL, SNW, WVP with 20m resolution, UTM 17N
+//     SUBDATASET_3_NAME=SENTINEL2_L2A:/vsizip/download/S2A.zip/S2A_MSIL2A_20160103T154242_N0201_R068_T17NRA_20160103T154241.SAFE/MTD_MSIL2A.xml:60m:EPSG_32617
+//     SUBDATASET_3_DESC=Bands B1, B9, AOT, CLD, SCL, SNW, WVP with 60m resolution, UTM 17N
+//     SUBDATASET_4_NAME=SENTINEL2_L2A:/vsizip/download/S2A.zip/S2A_MSIL2A_20160103T154242_N0201_R068_T17NRA_20160103T154241.SAFE/MTD_MSIL2A.xml:TCI:EPSG_32617
+//     SUBDATASET_4_DESC=True color image, UTM 17N
+impl Sentinel2Band {
+    pub fn resolution_meters(self) -> usize {
+        match self {
+            Sentinel2Band::B02 | Sentinel2Band::B03 | Sentinel2Band::B04 | Sentinel2Band::B08 => 10,
+            Sentinel2Band::B05
+            | Sentinel2Band::B06
+            | Sentinel2Band::B07
+            | Sentinel2Band::B08A
+            | Sentinel2Band::B11
+            | Sentinel2Band::B12 => 20,
+            Sentinel2Band::B01 | Sentinel2Band::B09 => 60,
+        }
+    }
+
+    #[allow(clippy::match_same_arms)]
+    pub fn channel_in_subdataset(self) -> usize {
+        match self {
+            Sentinel2Band::B01 => 1,
+            Sentinel2Band::B02 => 1,
+            Sentinel2Band::B03 => 2,
+            Sentinel2Band::B04 => 3,
+            Sentinel2Band::B05 => 1,
+            Sentinel2Band::B06 => 2,
+            Sentinel2Band::B07 => 3,
+            Sentinel2Band::B08 => 4,
+            Sentinel2Band::B08A => 4,
+            Sentinel2Band::B09 => 2,
+            Sentinel2Band::B11 => 5,
+            Sentinel2Band::B12 => 6,
+        }
+    }
+
+    #[allow(clippy::unused_self)] // might need to be used in the future to distinguish between bands
+    pub fn data_type(self) -> RasterDataType {
+        RasterDataType::U16
```

Comment:
SCL is u8 but thats not really a problem (yet)

Reply:
are you sure?

> Band 5 Block=128x128 Type=UInt16, ColorInterp=Undefined
  Description = SCL, Scene Classification
  Categories:
      0: NODATA
      1: SATURATED_DEFECTIVE
      2: DARK_FEATURE_SHADOW
      3: CLOUD_SHADOW
      4: VEGETATION
      5: NOT_VEGETATED
      6: WATER
      7: UNCLASSIFIED
      8: CLOUD_MEDIUM_PROBA
      9: CLOUD_HIGH_PROBA
     10: THIN_CIRRUS
     11: SNOW_ICE
  Metadata:
    BANDNAME=SCL

Reply:
well if it is not it should be :D however, you can keep it at iu6